pkg/hcp/telemetry/metrics.go

```diff
@@ -0,0 +1,246 @@
+package telemetry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	prompb "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+var (
+	errScraperFailedToCreateRequest = errors.New("failed to create request")
+	errScraperFailedToIssueRequest  = errors.New("failed to issue request")
+	errScraperFailedStatusCode      = errors.New("non-200 status code")
+	errScraperFailedToParseResponse = errors.New("failed to parse response")
+)
+
+type scraper interface {
+	scrape(ctx context.Context, filters []string, labels pcommon.Map) (pmetric.Metrics, error)
+}
+
+type scraperImpl struct {
+	client             http.RoundTripper
+	envoyAdminHostPort string
+	logger             hclog.Logger
+	parser             *expfmt.TextParser
+}
+
+func newScraper(envoyAdminHostPort string, logger hclog.Logger) scraper {
+	return &scraperImpl{
+		client:             http.DefaultTransport,
+		envoyAdminHostPort: envoyAdminHostPort,
+		logger:             logger,
+		parser:             new(expfmt.TextParser),
+	}
+}
+
+// scrape is called by the worker goroutine to scrape metrics from the Envoy admin API.
```

Comment:
```suggestion
// scrape is called by the exporter goroutine to scrape metrics from the Envoy admin API.
```

Reply:
agreed and makes sense. This and other references to workers were before a refactor/rename from worker > exporter. I just searched around for these remaining references.